### PR TITLE
Disallow quantifers on lookarounds and other zero-width assertion expressions

### LINKED ISSUE #68: Look-ahead/behind with repeat operator should not compile
From https://github.com/fancy-regex/fancy-regex/issues/66#issuecomment-779820378

The following types of regexes should not compile (but they currently do):

```
(?=a)*
(?!)?
(?=a){1,2}
(?<=a)*
(?<!a)*
```

Note that Oniguruma fails with this error: `target of repeat operator is invalid`

### PR DESCRIPTION
**Pull request by keith-hall**
Fixes #68 and matches Oniguruma behavior

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -172,6 +175,18 @@ impl<'a> Parser<'a> {
...
 
+    fn is_repeatable(&self, child: &Expr) -> bool {
+        match child {
+            Expr::LookAround(_, _) => false,
+            Expr::Empty => false,
+            Expr::StartText => false,
+            Expr::EndText => false,
+            Expr::StartLine => false,
+            Expr::EndLine => false,
+            _ => true
```
Comment (robinst) [inline src/parse.rs:186]: Mind running `cargo fmt`? Looks like the build is failing because of a missing comma here.

**Review by robinst — CHANGES_REQUESTED** [review]
Thanks! This looks good. Can you just rebase your changes on master and fix the formatting issue? Then we should have a clean build.

File: src/parse.rs (hunk 1 above)
Comment (keith-hall) [reply src/parse.rs]: done :) maybe worth mentioning this in the contributing guide to potentially save time in future for new contributors? :)

Comment (robinst) [reply src/parse.rs]: Yep done :)

**Pull request merged**